### PR TITLE
Truncate in middle for the dropdown of sorting columns in Experiment View

### PR DESCRIPTION
**Pull request by WeichenXu123**
Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

## What changes are proposed in this pull request?

Truncate in middle for the dropdown of sorting columns in Experiment View. Max length is 50 characters.

The behavior is:
image

## How is this patch tested?

N/A

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by dbczumar — COMMENTED** [review]
@WeichenXu123 This looks great! Can we fix merge conflicts with https://github.com/mlflow/mlflow/pull/5018 once it's merged? https://github.com/mlflow/mlflow/pull/5018 changes `width` to `auto` for the dropdown.

**Comment by WeichenXu123** [comment]
@dbczumar @harupy 
After I merged master, I found the command `mlflow ui` failed. Which commit break this ?

```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/bin/mlflow", line 33, in <module>
    sys.exit(load_entry_point('mlflow', 'console_scripts', 'mlflow')())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/weichen.xu/work/projects/mlflow/mlflow/cli.py", line 280, in ui
    _run_server(backend_store_uri, default_artifact_root, host, port, None, 1)
  File "/Users/weichen.xu/work/projects/mlflow/mlflow/server/__init__.py", line 142, in _run_server
    exec_cmd(full_command, env=env_map, stream_output=True)
  File "/Users/weichen.xu/work/projects/mlflow/mlflow/utils/process.py", line 35, in exec_cmd
    cmd, env=cmd_env, cwd=cwd, universal_newlines=True, stdin=subprocess.PIPE, **kwargs
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 1462, in _execute_child
    env_list.append(k + b'=' + os.fsencode(v))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/os.py", line 812, in fsencode
    filename = fspath(filename)  # Does type-checking of `filename`.
TypeError: expected str, bytes or os.PathLike object, not int
```

**Comment by harupy** [comment]
Hey @WeichenXu123, https://github.com/mlflow/mlflow/pull/5032 should fix the issue.

**Review by dbczumar — APPROVED** [review]
LGTM!

File: mlflow/server/js/src/common/utils/StringUtils.js
Comment (harupy) [inline mlflow/server/js/src/common/utils/StringUtils.js:29]:
```suggestion
export const middleTruncate = (s, maxLen) => {
  if (keyName.length > maxLen) {
    const firstPartLen = Math.floor((maxLen - 3) / 2);
    const lastPartLen = maxLen - 3 - firstPartLen;
    return (
      keyName.substring(0, firstPartLen) +
      '...' +
      keyName.substring(keyName.length - lastPartLen, keyName.length)
    );
  } else {
    return keyName;
  }
};
```

Can we use a more generic name here because this function can be applied any string?

File: mlflow/server/js/src/common/utils/StringUtils.js
Comment (harupy) [inline mlflow/server/js/src/common/utils/StringUtils.js:17]:
```suggestion
export const middleTruncate = (str, maxLen) => {
```

Comment (harupy) [reply mlflow/server/js/src/common/utils/StringUtils.js:17]: or do you prefer `middleTruncateKey`?

**Review by harupy — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/5022/files#r746286885 is addressed

File: mlflow/server/js/src/common/utils/StringUtils.js
Comment (WeichenXu123) [reply mlflow/server/js/src/common/utils/StringUtils.js:17]: middleTruncateStr looks better

Comment (harupy) [reply mlflow/server/js/src/common/utils/StringUtils.js]: `middleTruncateStr` sounds good to me :)

**Pull request merged**
